src/Analyser/NodeScopeResolver.php

```diff
@@ -370,16 +378,16 @@ public function processStmtNodes(
 			$throwPoints = array_merge($throwPoints, $statementResult->getThrowPoints());
 			$impurePoints = array_merge($impurePoints, $statementResult->getImpurePoints());
 
-			if (!$statementResult->isAlwaysTerminating()) {
+			if ($alreadyTerminated || !$statementResult->isAlwaysTerminating()) {
 				continue;
 			}
 
 			$alreadyTerminated = true;
 			$nextStmt = $this->getFirstUnreachableNode(array_slice($stmts, $i + 1), $parentNode instanceof Node\Stmt\Namespace_);
-			if ($nextStmt !== null) {
-				$nodeCallback(new UnreachableStatementNode($nextStmt), $scope);
+			if ($nextStmt === null) {
+				continue;
 			}
-			break;
+			$nodeCallback(new UnreachableStatementNode($nextStmt), $scope);
```

Comment:
Sorry, I don't understand these changes. So here we're going to invoke `UnreachableStatementNode` but then we're still going to analyse the next statement if it's not Function_ or ClassLike? I don't get it.

Reply:
Given this code:
```php
<?php
namespace Foo;

doSomething();
exit();
foo();
bar();

function doSomething(): void
{
  // ...
}
```

Here `foo();` will be reported as an unreachable statement.
Before this PR, PHPStan stops analyzing statements after `foo()` because analyzing `bar()` is no longer useful.
However, we still need to analyze the body of `doSomething()` actually.

So PHPStan should continue analysis after an unreachable statement, but the target is different;
- Before unreachable statements (`$alreadyTerminated = false`)
  - analyze all statements
- After unreachable statements (`$alreadyTerminated = true`)
  - analyze only functions and class-likes

Reply:
what about using `PhpParser\Node\FunctionLike` to also cover methods/closures, arrow-functions?

Reply:
@staabm 1) I don't think you can put `exit` in-between methods in a class
2) I don't think closures after `exit` are executed.

Reply:
@takaram Thank you, makes total sense!